packages/compass-import-export/src/utils/collection-stream.ts

```diff
@@ -223,7 +227,22 @@ export class WritableCollectionStream extends Writable {
 
     this.emit('progress', progressStats);
 
-    return callback();
+    return callback(this._makeStreamError());
+  }
+
+  _makeStreamError(): CollectionStreamError | undefined {
+    if (this.stopOnErrors && this._errors.length) {
+      const error = this._errors[0];
+      if (error instanceof Error) {
+        return error;
```

Comment:
```suggestion
      if (Object.prototype.toString.call(error) === '[object Error]') {
        return error as Error;
```

?

Reply:
Sure, but I do wonder if/how that could ever break.